loader/loader.c

```diff
@@ -5133,6 +5172,39 @@ VkResult loader_validate_device_extensions(struct loader_instance *this_instance
     return VK_SUCCESS;
 }
 
+static VkResult loader_get_layer_required_instance_extensions(uint32_t *pRequiredExtraInstanceExtensionCount, char **pRequiredExtraInstanceExtensions) {
+    struct loader_layer_list instance_layers;
+    uint32_t required_inst_ext_count = 0;
+    VkResult res = VK_SUCCESS;
+
+    memset(&instance_layers, 0, sizeof(instance_layers));
+
+    res = loader_scan_for_layers(NULL, &instance_layers);
```

Comment:
loader_scan_for_layers is a BIG function, and I'd much rather not have it be called a second time in `VkCreateInstance` (here in the terminator). It *shouldn't* return a different list, and instead it should be possible to do all the logic required here before calling down the vkCreateInstance chain. 

What I imagine is that after loader_scan_for_layers is called in vkCreateInstance, create a new list of extension names to enable which contains all the extensions from the application and any required extensions from layers.
The loader replaces function parameters in numerous places for similar reasons.